contracts/operators/Beefy/BeefyVaultStorage.sol

```diff
@@ -0,0 +1,35 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity 0.8.11;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+/// @title BeefyVaultOperator storage contract
+contract BeefyVaultStorage is Ownable {
+    /// @dev Emitted when a vault is added
+    /// @param vault The vault address
+    /// @param token The underlying token address
+    event VaultAdded(address vault, address token);
+
+    /// @dev Emitted when a vault is removed
+    /// @param vault The removed vault address
+    event VaultRemoved(address vault);
+
+    /// @dev Map of vault address with underlying token address
+    mapping(address => address) public vaults;
+
+    /// @notice Add a beefy single asset vault
+    /// @param vault The vault address
+    /// @param token The underlying token address (used to deposit)
+    function addVault(address vault, address token) external onlyOwner {
+        require(vault != address(0), "BVS: INVALID_VAULT_ADDRESS");
+        require(token != address(0), "BVS: INVALID_TOKEN_ADDRESS");
+        vaults[vault] = token;
```

Comment:
This method allowes to override the current token address for a given vault.
We could add a require to check that the vault does not exist yet, enforcing the owner to delete a vault and re-add it if the token address changes.
In most cases, it can prevent overriding by mistake.